automation/label-approved-veps.py

```diff
@@ -0,0 +1,96 @@
+
+def get_pr_details():
+    #Fetch the kubevirt/kuebvirt PR body
+    url = f"https://api.github.com/repos/{KUBEVIRT_REPO}/pulls/{PR_NUMBER}"
+    response = requests.get(url, headers=HEADERS)
+    response.raise_for_status()
+    return response.json()["body"]
+
+def extract_enhancements_references(pr_body):
+    #Get the enhancements reference numbers from the PR.
+    pattern = r"(?:https://github.com/)?kubevirt/enhancements/(?:issues|pull)/(\d+)|(?:kubevirt/)?enhancements#(\d+)"
+    matches = re.findall(pattern, pr_body)
+
+    # Extract the first non-empty group from each match
+    ref_numbers = [group[0] or group[1] for group in matches]
+    # there can be duplicates
+    return list(set(ref_numbers))
+
+def is_merged_pr(ref_number):
+    url = f"https://api.github.com/repos/kubevirt/enhancements/pulls/{ref_number}"
+    response = requests.get(url, headers=HEADERS)
+    if response.status_code == 200:
+        pr_data = response.json()
+        return pr_data.get("merged_at") is not None
+    return False
+
+def find_related_merged_prs(issue_number):
+    url = "https://api.github.com/repos/kubevirt/enhancements/pulls?state=closed&per_page=100"
+    try:
+        response = requests.get(url, headers=HEADERS)
+        if response.status_code != 200:
+            print(f"API error: {response.status_code} - {response.text}")
+            return []
+        
+        prs = response.json()
+        related_prs = []
+        for pr in prs:
+            # Check if PR is merged and references the issue in its body
+            # that's a bit of a mess since there can be multiple ways to reference 
+            body = pr.get('body', '')
+            if (pr['merged_at'] and 
+                (f"#{issue_number}" in body or 
+                 f"issues/{issue_number}" in body or 
+                 f"https://github.com/kubevirt/enhancements/issues/{issue_number}" in body)):
+                related_prs.append(pr['number'])
+        return related_prs
+    
+    except Exception as e:
+        print(f"Request failed: {e}")
+        return []
+
+def add_label_to_pr():
+    # Adds the 'approved-vep' label to the kubevirt PR. (when we will have that )
+    url = f"https://api.github.com/repos/{KUBEVIRT_REPO}/issues/{PR_NUMBER}/labels"
+    payload = {"labels": ["approved-vep"]}
+    response = requests.post(url, headers=HEADERS, json=payload)
+    response.raise_for_status()
+
+def main():
+    # Get PR body
+    pr_body = get_pr_details()
+    if not pr_body:
+        print("No PR body found.")
+        return
+
+    # Extract enhancements reference numbers
+    ref_numbers = extract_enhancements_references(pr_body)
+    if not ref_numbers:
+        print("No enhancements references found.")
+        return
+
+    # Check the first reference
+    ref_number = ref_numbers[0]
+    if is_merged_pr(ref_number):
+        print(f"Referenced PR enhancements#{ref_number} is merged.")
+        add_label_to_pr()
+    else:
```

Comment:
I don't understand this else... Could use more comments.
If we reference a merged enhancement we add the approved-vep label.
If not, we look at all the enhancements hoping to find the referenced enhancement mentioned there? Just its number? I'm confused...

Reply:
yeah. I thought that the VEP PR can be referenced directly or a VEP issue.
If it's not a VEP PR then I assume it's an issue and am looking for any merged PRs referenced by this issue. 
I can drop the `is_merged_pr` altogether and force users to reference only the issue. 
What do you think?